core/src/main/kotlin/com/tschuchort/compiletesting/JavacUtils.kt

```diff
@@ -59,48 +57,13 @@ internal class StringJavaFileObject(className: String, private val contents: Str
     override fun getLastModified(): Long = _lastModified
 }
 
-/**
- * Gets the version string of a javac executable that can be started using the
- * given [javacCommand] via [Runtime.exec].
- */
-internal fun getJavacVersionString(javacCommand: String): String {
-    val javacProc = ProcessBuilder(listOf(javacCommand, "-version"))
-        .redirectErrorStream(true)
-        .start()
-
-    val buffer = Buffer()
-
-    javacProc.inputStream.copyTo(buffer.outputStream())
-    javacProc.waitFor()
-
-    val output = buffer.readUtf8()
-
-    return parseVersionString(output) ?: throw IllegalStateException(
-        "Command '$javacCommand -version' did not print expected output. " +
-                "Output was: '$output'"
-    )
+internal fun isJavac9OrLater(): Boolean {
+    return isJavac9OrLater(System.getProperty("java.specification.version"))
 }
 
-internal fun parseVersionString(output: String) =
-    Regex("javac (.*)?[\\s\\S]*").find(output)?.destructured?.component1()
-
-internal fun isJavac9OrLater(javacVersionString: String): Boolean {
-    try {
-        val (majorv, minorv, patchv, otherv) = Regex("([0-9]*)(?:\\.([0-9]*))?(?:\\.([0-9]*))?(.*)")
-            .matchEntire(javacVersionString)?.destructured
-            ?: throw IllegalArgumentException("Could not match version regex")
-
-        check(majorv.isNotBlank()) { "Major version can not be blank" }
-
-        if (majorv.toInt() == 1)
-            check(minorv.isNotBlank()) { "Minor version can not be blank if major version is 1" }
-
-        return (majorv.toInt() == 1 && minorv.toInt() >= 9) // old versioning scheme: 1.8.x
-                || (majorv.toInt() >= 9) // new versioning scheme: 10.x.x
-    }
-    catch (t: Throwable) {
-        throw IllegalArgumentException("Could not parse javac version string: '$javacVersionString'", t)
-    }
+internal fun isJavac9OrLater(specVersion: String): Boolean {
+    val version = specVersion.toIntOrNull() ?: return false
+    return version >= 9
```

Comment:
This will no longer handle the case where a JDK 9 still uses the old versioning scheme (1.9.0). Though I don't know if that can ever happen. Maybe some very early access builds do it, but I would be happy to ignore them.

Reply:
I don't think that needs to be supported